spec/models/team_spec.rb

```diff
@@ -233,18 +293,43 @@
       context 'when handle_dups option is rename' do
         it 'returns new team name' do
           allow(Course).to receive(:find).with(1).and_return(double('Course', name: 'no course'))
+          allow(Assignment).to receive(:find).with(1).and_return(double('Assignment', name: 'no assignment'))
           allow(Team).to receive(:generate_team_name).with('no course').and_return('new team name')
+          allow(Team).to receive(:generate_team_name).with('no assignment').and_return('new team name')
           expect(Team.handle_duplicate(team, 'no name', 1, 'rename', CourseTeam.new)).to eq('new team name')
+          expect(Team.handle_duplicate(team, 'no name', 1, 'rename', AssignmentTeam.new)).to eq('new team name')
         end
       end
 
       context 'when handle_dups option is replace' do
         it 'deletes the old team' do
-          allow(TeamsUser).to receive_message_chain(:where, :find_each).with(team_id: 1).with(no_args).and_yield(team_user)
-          allow(team_user).to receive(:destroy).and_return(team_user)
+          allow(team).to receive(:delete)
           expect(Team.handle_duplicate(team, 'no name', 1, 'replace', CourseTeam.new)).to eq('no name')
         end
       end
+
+      context 'when handle_dups option is insert' do
+        it 'does nothing and returns nil' do
+          expect(Team.handle_duplicate(team, 'no name', 1, 'insert', CourseTeam.new)).to be nil
+        end
+      end
+
+      # By the time this test is added (by E1949), the renaming existing team function does not exist yet,
+      # so it should fail unless the function is implemented and the existing team is renamed and saved.
```

Comment:
The Travis CI checks failed because it was supposed to fail. Check the commented lines.